include/ImmediateContext.inl

```diff
@@ -263,14 +263,22 @@ inline bool CViewBoundState<TBindable, NumBindSlots>::IsDirty(TDeclVector const&
         }
     }
 
-    if (!bDirty)
+    if (m_ShaderData.size() < rootSignatureBucketSize)
+    {
+        // Did we move to a larger bucket size? If so, fill the extra shader data to null (unknown) descriptors
+        m_ShaderData.resize(rootSignatureBucketSize, c_AnyNull);
+        bDirty = true;
+    }
+    else if (m_ShaderData.size() > rootSignatureBucketSize)
     {
-        bDirty = DirtyBitsUpTo(static_cast<UINT>(New.size()));
+        //Did we move to a smaller bucket size? If so, shrink the shader data to fit
+        m_ShaderData.resize(rootSignatureBucketSize);
+        bDirty = true;

```

Comment:
This doesn't need to mark the range as dirty, we don't need to copy new descriptors in this case. There can be stale descriptors from the old table sitting around, as long as the root signature can't address them.

Reply:
Oh true, that makes sense. Thanks!